### PR TITLE
Add support for previewing PDF attachments

### LINKED ISSUE #2800: consider previewing pdfs with pdf.js
> Would it eventually be possible to preview pdfs? :) browsers should be able to do it

Yes, but I didn't manage to make pdf preview with native ways (iframe/embed/object) so the solution would probably require 3rd-party solution like Mozilla's [pdf.js](https://github.com/mozilla/pdf.js)

_Originally posted by @limonte in https://github.com/FlowCrypt/flowcrypt-browser/diffs_

### PR DESCRIPTION
**Pull request by limonte**
Closes #2800

### CODE REVIEW

**Comment by limonte** [comment]
@tomholub this is the initial work on PDF preview, please check it out and let me know what controls you'd like to have there.

IMO Gmails controls are simple and cover all the need users have when previewing PDF files:

image

We can re-create the same toolbar if you think it's needed.

**Review by tomholub — COMMENTED** [review]
This is absolutely amazing - things like this is why I'm happy to have you be involved :+1:

Agree with the controls.

Separately, I noticed (on chrome, enterprise version build) that the download button on top of the preview doesn't work (for any file type). The download button in the mini attachment frame works. I haven't tested if it's related to this PR, or if it happens on master too. Please also have a look and add tests in case it's not tested.

File: build.sh [hunk 1]
```diff
@@ -58,6 +58,8 @@ cp node_modules/squire-rte/build/squire-raw.js $OUTDIR/lib/squire-raw.js
 cp node_modules/clipboard/dist/clipboard.js $OUTDIR/lib/clipboard.js
 cp node_modules/@flowcrypt/fine-uploader/fine-uploader/fine-uploader.js $OUTDIR/lib/fine-uploader.js
 cp node_modules/filesize/lib/filesize.js $OUTDIR/lib/filesize.js
+cp node_modules/pdfjs-dist/build/pdf.js $OUTDIR/lib/pdf.js
+cp node_modules/pdfjs-dist/build/pdf.worker.js $OUTDIR/lib/pdf.worker.js
```
[the remark this answers is not in the document]
Comment (limonte) [reply build.sh:62]: Yes, it's used by `pdf.js`: 

image

**Comment by limonte** [comment]
@tomholub could you please help me with mocking the "file preview - open txt file, image, pdf, unknown" message with `threadId` `174ab0ba9643b4fa`?

I'll now switch to https://github.com/FlowCrypt/flowcrypt-browser/issues/2950 because it seems critical for me.

**Comment by tomholub** [comment]
For adding mock, you can newly follow these instructions:
```
export class GoogleData {

  /**
   * This is the proper way to add messages to mock api for testing:
   *   1) log into flowcrypt.compatibility@gmail.com
   *   2) go to Encrypted Inbox and find your message
   *   3) click "download api export"
   *   4) save the json file to exported-messages folder
   */
  private exportedMsgsPath = './test/source/mock/google/exported-messages/';
```

**Comment by limonte** [comment]
@tomholub I can't find the "download api export" button mentioned in step 3. When is it located?

File: test/source/mock/google/google-data.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ export class GoogleData {
   /**
    * This is the proper way to add messages to mock api for testing:
    *   1) log into flowcrypt.compatibility@gmail.com
-   *   2) go to Encrypted Inbox and find your message
+   *   2) go to Settings -> Inbox and find your message
```
Comment (tomholub) [inline test/source/mock/google/google-data.ts:93]: :+1:

**Comment by limonte** [comment]
@tomholub for the sake of smaller PRs, would you agree to merge this one and implement controls in another PR? I'll the create the separated task for PDF viewer controls which I'll start after #2951 because #2951 is more important to tackle first.

**Comment by tomholub** [comment]
100% agree

**Review by tomholub — APPROVED** [review]

**Pull request merged**